cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -71,6 +71,25 @@ func cleanup(indexDir string, repos []string, now time.Time) {
 
 		moveAll(trashDir, shards)
 	}
+
+	// Remove old .tmp files from crashed indexer runs-- for example, if
+	// an indexer OOMs, it will leave around .tmp files, usually in a loop.
+	maxAge := now.Add(-4 * time.Hour)
+	if failures, err := filepath.Glob(filepath.Join(indexDir, "*.tmp")); err != nil {
+		log.Printf("Glob: %v", err)
+	} else {
+		for _, f := range failures {
+			st, err := os.Stat(f)
+			if err != nil {
+				log.Printf("Stat(%q): %v", f, err)
+				continue
+			}
+			if !st.IsDir() && st.ModTime().Before(maxAge) {
+				log.Printf("removing old tmp file: %s", f)
+				os.Remove(f)
```

Comment:
I think we should print errors returned by this call if non-nil. It could reveal some unexpected issues way faster when debugging. It would come in handy at the right time we need it (to avoid editing code on rush and cut a patch release).

Reply:
If this errors out, you'll see repeated "removing old tmp file" messages. But errors would be pretty strange-- you'd need the zoekt directory to not be writable by the indexserver to not be able to delete a file in it, or some other exotic situation that would cause such a broken state that nothing else would work.

Reply:
>repeated "removing old tmp file" messages

That could also work, just a simple recuse for the future self 😁